src/plugins/discover/public/utils/on_resize_grid_column.ts

```diff
@@ -9,13 +9,13 @@
 import type { DiscoverGridSettings } from '@kbn/saved-search-plugin/common';
 
 export const onResizeGridColumn = (
-  colSettings: { columnId: string; width: number },
+  colSettings: { columnId: string; width?: number },
```

Comment:
nit to make it more explicit that the width is not missing rather needs to be reset.
We could also leave a comment explaining the logic.

```suggestion
  colSettings: { columnId: string; width: number | undefined },
```

Reply:
Agreed 👍 Updated here and added comments explaining why it can be `undefined`: https://github.com/elastic/kibana/pull/190288/commits/984453586b85e4354b3e53774d7481f2be43db67